provider/aci.go

```diff
@@ -1632,9 +1638,59 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*aci.ContainerProbe, e
 	}, nil
 }
 
+func (p *ACIProvider) getAzureFileCSI(volume v1.Volume, namespace string) (*aci.Volume, error) {
+	azureSource := &aci.AzureFileVolume{
+		ReadOnly: *volume.CSI.ReadOnly,
+	}
+	if volume.CSI.NodePublishSecretRef != nil && volume.CSI.NodePublishSecretRef.Name != "" {
+
+		// Set the storage account name and key
+		secret, err := p.resourceManager.GetSecret(volume.CSI.NodePublishSecretRef.Name, namespace)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("The secret %s for AzureFile CSI driver %s is not found", volume.CSI.NodePublishSecretRef.Name, volume.Name))
+		}
+
+		if secret == nil {
+			return nil, fmt.Errorf("Getting secret for AzureFile CSI driver %s returned an empty secret", volume.Name)
+		}
+
+		azureSource.StorageAccountName = string(secret.Data["azurestorageaccountname"])
+		azureSource.StorageAccountKey = string(secret.Data["azurestorageaccountkey"])
+
+		// Set shareName
+		if volume.CSI.VolumeAttributes != nil {
+			if shareName, ok := volume.CSI.VolumeAttributes[azureFileShareName]; ok {
+				azureSource.ShareName = shareName
+			}
+		}
+		volume := aci.Volume{
+			Name:      volume.Name,
+			AzureFile: azureSource,
+		}
+		return &volume, nil
+	} else {
+		return nil, fmt.Errorf("NodePublishSecretRef for AzureFile CSI driver %s cannot be empty or nil", volume.Name)
+	}
+}
+
 func (p *ACIProvider) getVolumes(pod *v1.Pod) ([]aci.Volume, error) {
 	volumes := make([]aci.Volume, 0, len(pod.Spec.Volumes))
 	for _, v := range pod.Spec.Volumes {
+		// Handle the case for Azure File CSI driver
+		if v.CSI != nil {
+			// Check if the CSI driver if file (Disk is not supported by ACI)
+			if v.CSI.Driver == AzureFileDriverName {
+				csiVolume, err := p.getAzureFileCSI(v, pod.Namespace)
+				if err != nil {
+					return volumes, err
```

Comment:
why return volumes here but not at line 1690?

Reply:
fixed to be `nil`